token_dispenser_client/token_dispenser_client.py

```diff
@@ -0,0 +1,177 @@
+            if result_count == 2:
+                raise ValueError(f"Found multiple parameters in path: {ssm_name}. Please provide "
+                                 f"specific ssm name which points to the TDS lambda ARN. "
+                                 f"Not path")
+            logger.debug('Found single ssm param value by default ssm path')
+            param = response.get('Parameters', []).pop(0)
+            return param['Value']
+        except KeyError as key_exception:
+            logger.exception('ssm get_parameters_by_path error with path:%s '
+                             'response:%s', ssm_name, response)
+            raise key_exception
+        except Exception as general_exception:
+            logger.exception('Unexpected error in get_parameters_by_path with '
+                             'path:%s response:%s', ssm_name, response)
+            raise general_exception
+    # if user provides a name, the code trusts it as a full name and let system fail if
+    # provided name is not correct
+    return get_parameter_by_name(name=ssm_name)
+
+
+def get_token(client_id: str, minimum_alive_secs: int = 300,
+              token_dispenser_arn_ssm_key: str = None) -> str:
+    """
+    Retrieves parameters from AWS Systems Manager Parameter Store by path.
+
+    Args:
+        client_id (str): Caller defined client_id
+        minimum_alive_secs (int, optional): minimum alive secs for the token. Defaults to 300.
+        If there is a cached token and the cached token has expiration time
+        (current_time - expired_at)  shorter than the provided value,
+        the token will be re-generated. Otherwise, a cached token will be returned.
+        token_dispenser_arn_ssm_key (str, optional): If not provided, the program
+        logic will /service/token-dispenser as root name space to get parameters.
+        If only single value is returned, the returned value will be used as
+        token dispenser arn.  Furthermore, the code will make a call to the token dispenser
+        based on the retrieved ARN.
+
+    Returns:
+        str: A json string containing launchpad token with other fields such as created_at
+        and expired_at all in EPOCH format
+
+        Returns empty dict if no parameters found on the path.
+            The return string could be a string showing error message
+    """
+    # Validate inputs.   minimum_alive_secs has a default value and won't be None
+    err_msgs: List = validate_input(client_id=client_id,
+                                    minimum_alive_secs=minimum_alive_secs)
+    if len(err_msgs) > 0:
+        raise ValueError(err_msgs)
+
+    # TDS stands for Token Dispenser Service (it is a lambda)
+    tds_arn = get_tds_arn(token_dispenser_arn_ssm_key)
+    data = {
+        "client_id": client_id,
+        "minimum_alive_secs": minimum_alive_secs
+    }
+    json_data = json.dumps(data)
+    resp: str = invoke_lambda(input_params_json=json_data, lambda_arn=tds_arn)
+    return resp
```

Comment:
`json.loads` the response from the lambda so we don't have to deserialize it upstream